src/main/kotlin/no/nav/familie/ef/mottak/hendelse/JournalfoeringHendelseDbUtil.kt

```diff
@@ -0,0 +1,57 @@
+package no.nav.familie.ef.mottak.hendelse
+
+import no.nav.familie.ef.mottak.repository.HendelsesloggRepository
+import no.nav.familie.ef.mottak.repository.domain.Hendelseslogg
+import no.nav.familie.ef.mottak.task.LagEksternJournalføringsoppgaveTask
+import no.nav.familie.kontrakter.felles.journalpost.Journalpost
+import no.nav.familie.log.IdUtils
+import no.nav.familie.log.mdc.MDCConstants
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.prosessering.domene.TaskRepository
+import no.nav.joarkjournalfoeringhendelser.JournalfoeringHendelseRecord
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import org.springframework.stereotype.Service
+import java.util.Properties
+import javax.transaction.Transactional
+
+@Service
+class JournalfoeringHendelseDbUtil(val hendelseloggRepository: HendelsesloggRepository,
+                                   val taskRepository: TaskRepository) {
+
+    val logger: Logger = LoggerFactory.getLogger(JournalfoeringHendelseDbUtil::class.java)
+
+    @Transactional
+    fun lagreHendelseslogg(hendelseRecord: JournalfoeringHendelseRecord, offset: Long): Hendelseslogg {
+        return hendelseloggRepository
+                .save(Hendelseslogg(offset,
+                                    hendelseRecord.hendelsesId.toString(),
+                                    mapOf("journalpostId" to hendelseRecord.journalpostId.toString(),
+                                          "hendelsesType" to hendelseRecord.hendelsesType.toString()).toProperties()))
+    }
+
+    fun erHendelseRegistrertIHendelseslogg(hendelseRecord: JournalfoeringHendelseRecord) =
+            hendelseloggRepository.existsByHendelseId(hendelseRecord.hendelsesId.toString())
+
+    fun lagreEksternJournalføringsTask(journalpost: Journalpost) {
+        logger.info("Oppretter task LagEksternJournalføringsoppgaveTask, feature skrudd på")
+        val metadata = opprettMetadata(journalpost)
+        val journalføringsTask = Task(type = LagEksternJournalføringsoppgaveTask.TYPE,
+                                      payload = journalpost.journalpostId,
+                                      metadata = metadata)
+        taskRepository.save(journalføringsTask)
+    }
+
+    private fun opprettMetadata(journalpost: Journalpost): Properties {
+        return Properties().apply {
+            if (journalpost.bruker != null) {
+                this["personIdent"] = journalpost.bruker!!.id
```

Comment:
```
journalpost.bruker?.let {
    this["personIdent"] = it.id
}
```

Slipper vi å ha `!!` bak bruker?

Reply:
Evt `this["personIdent"] = journalpost.bruker?.id ?: null`

Reply:
Enig, det er bra å unngå `!!`

Testet første løsningsforslag og fikk:
Smart cast to 'Bruker' is impossible, because 'journalpost.bruker' is a public API property declared in different module
Skjønte tilbakemeldingen litt mer etter at jeg leste den her: https://discuss.kotlinlang.org/t/what-is-the-reason-behind-smart-cast-being-impossible-to-perform-when-referenced-class-is-in-another-module/2201

Andre løsning blir vel egentlig det samme som å skrive:
`this["personIdent"] = journalpost.bruker?.id`

Liker andre løsningsforslag godt, men ble litt usikker i forhold til properties hvis jeg gjør den endringen,
om det blir noe forskjell funksjonelt på å ha en property (personIdent, null) og ikke ha den propertyen i det hele tatt.
Skal undersøke det.

Reply:
I første løsning får du den feilen hvis du bruker `journalpost.bruker.id` inni let, men med `it.id` så trodde jeg det skulle gå fint?

Tror ikke vi bruker metadataene personIdent til noe annet enn logging og visning i https://familie-prosessering.dev.nav.no/service/familie-ef-mottak/gruppert